src/raster/dataset.rs

```diff
@@ -49,11 +48,22 @@ impl Dataset {
 
     pub unsafe fn _c_ptr(&self) -> *const c_void {
         return self.c_dataset;
-    }
-
-    pub fn size(&self) -> (isize, isize) {
-        let size_x = unsafe { gdal::GDALGetRasterXSize(self.c_dataset) } as isize;
-        let size_y = unsafe { gdal::GDALGetRasterYSize(self.c_dataset) } as isize;
+    }
+
+
+    pub fn get_rasterband<'a>(&'a self, band_index: isize) -> Option<RasterBand<'a>> {

```

Comment:
Maybe that's the price for working with multiple languages at the same time ;)  I will remove all `get_`prefixes.